src/coreComponents/constitutive/solid/DruckerPragerExtended.hpp

```diff
@@ -0,0 +1,495 @@
+                   m_oldState[k][q],
+                   friction,
+                   dfriction_dstate );
+
+  real64 yield = trialQ + friction * (trialP - m_pressureIntercept[k]);
+
+  if( yield < 1e-9 ) // elasticity
+  {
+    return;
+  }
+
+  // else, plasticity (trial stress point lies outside yield surface)
+  // the return mapping can in general be written as a newton iteration.
+
+  real64 solution[3], residual[3], delta[3];
+  real64 jacobian[3][3] = {{}}, jacobianInv[3][3] = {{}};
+
+  solution[0] = trialP; // initial guess for newP
+  solution[1] = trialQ; // initial guess for newQ
+  solution[2] = 1e-5;   // initial guess for plastic multiplier
+
+  real64 norm, normZero = 1e30;
+
+  // begin newton loop
+
+  for( localIndex iter=0; iter<20; ++iter )
+  {
+    m_newState[k][q] = m_oldState[k][q] + solution[2];
+
+    hyperbolicModel( m_initialFriction[k],
+                     m_residualFriction[k],
+                     m_hardening[k],
+                     m_newState[k][q],
+                     friction,
+                     dfriction_dstate );
+
+    // assemble residual system
+    // resid1 = P - trialP + dlambda*bulkMod*dG/dP = 0
+    // resid2 = Q - trialQ + dlambda*3*shearMod*dG/dQ = 0
+    // resid3 = F = 0
+
+    residual[0] = solution[0] - trialP + solution[2] * m_bulkModulus[k] * m_dilationRatio[k] * friction;
+    residual[1] = solution[1] - trialQ + solution[2] * 3 * m_shearModulus[k];
+    residual[2] = solution[1] + friction * (solution[0] - m_pressureIntercept[k]);
+
+    // check for convergence
+
+    norm = LvArray::tensorOps::l2Norm< 3 >( residual );
+
+    if( iter==0 )
+    {
+      normZero = norm;
+    }
+
+    if( norm < 1e-8*(normZero+1))
+    {
+      break;
+    }
+
+    // solve Newton system
+
+    jacobian[0][0] = 1;
+    jacobian[0][2] = m_bulkModulus[k] * m_dilationRatio[k] * friction + solution[2] * m_bulkModulus[k] * m_dilationRatio[k] * dfriction_dstate;
+    jacobian[1][1] = 1;
+    jacobian[1][2] = 3 * m_shearModulus[k];
+    jacobian[2][0] = friction;
+    jacobian[2][1] = 1;
+    jacobian[2][2] = dfriction_dstate * (solution[0]-m_pressureIntercept[k]);
+
+    LvArray::tensorOps::invert< 3 >( jacobianInv, jacobian );
+    LvArray::tensorOps::Ri_eq_AijBj< 3, 3 >( delta, jacobianInv, residual );
+
+    for( localIndex i=0; i<3; ++i )
+    {
+      solution[i] -= delta[i];
+    }
+  }
+
+  // re-construct stress = P*eye + sqrt(2/3)*Q*nhat
+
+  twoInvariant::stressRecomposition( solution[0],
+                                     solution[1],
+                                     deviator,
+                                     stress );
+
+  // construct consistent tangent stiffness
+  // note: if trialQ = 0, we will get a divide by zero error below,
+  // but this is an unphysical (zero-strength) state anyway
+
+  LvArray::tensorOps::fill< 6, 6 >( stiffness, 0 );
+
+  real64 c1 = 2 * m_shearModulus[k] * solution[1] / trialQ;
+  real64 c2 = jacobianInv[0][0] * m_bulkModulus[k] - c1 / 3;
```

Comment:
We may have the case where trialQ equal to zero, e.g. isotropic loading condition. This may cause the division to zero issue here.